test/core_ext/deep_prune_test.rb

```diff
@@ -0,0 +1,63 @@
+# encoding: utf-8
+
+require File.expand_path(File.join(File.dirname(__FILE__) + '/../test_helper'))
+require File.expand_path(File.join(File.dirname(__FILE__) + '/../../lib/core_ext/hash/deep_prune'))
+
+class TestDeepprune < Test::Unit::TestCase
+  test "deep_prune! returns empty Hash for empty Hash" do
+    assert_equal({}, {}.deep_prune!)
+  end
+
+  test "deep_prune! with default comparator calls deep_prune! recursively and removes empty hashes" do
+    hash = {
+      "nil_value" => nil,
+      "empty_hash" => {},
+      "nested_empty_hash" => {
+        "foo" => {},
+      },
+      "nested_non_empty_hash" => {
+        "bar" => "baz",
+      },
+      "empty_string" => "",
+      "empty_array" => [],
+      "integer" => 1,
+    }
+
+    expected = {
+      "nil_value" => nil,
+      "nested_non_empty_hash" => {
+        "bar" => "baz",
+      },
+      "empty_string" => "",
+      "empty_array" => [],
+      "integer" => 1,
+    }
+    assert_equal(expected, hash.deep_prune!)
+  end
+
+  test "deep_prune! can be called with other comparators to removes blank items" do
+    blank = ->(v) { v.respond_to?(:empty?) ? !!v.empty? : !v } # Rails' Object#blank?
+
+    hash = {
+      "nil_value" => nil,
+      "empty_hash" => {},
+      "nested_empty_hash" => {
+        "foo" => {},
+      },
+      "nested_non_empty_hash" => {
+        "bar" => "baz",
```

Comment:
If you add a nested empty string here, will it be removed?
I'm guessing not because of [`if v.is_a?(Hash)`](https://github.com/ruby-i18n/ruby-cldr/pull/86/files#diff-3fa47c2ea0092e19ded5bb3696b8db105a0a45c03276a4783ca5819721e9baa2R4).
```suggestion
        "bar" => "baz",
        "empty_string" => "",
```

Reply:
> If you add a nested empty string here, will it be removed?

Yes. In this test it will be. In the other test it won't.

The `if v.is_a?(Hash)` just determines whether we recurse, not what we prune.